entity-framework/core/logging-events-diagnostics/event-counters.md

```diff
@@ -0,0 +1,84 @@
+
+### [Windows](#tab/windows)
+
+1. Open the Windows Task Manager by right-clicking on the task bar and selecting "Task Manager.
+2. Make sure that the "More details" option is selected at the bottom of the window.
+3. In the Processes tab, right-click a column and make sure that the PID column is enabled.
+4. Locate your application in the process list, and get its process ID from the PID column.
+
+### [Linux or macOS](#tab/fluent-api)
+
+1. Use the `ps` command to list all processes running for your user.
+2. Locate your application in the process list, and get its process ID from the PID column.
+
+***
+
+Inside your .NET application, the process ID is available as `Process.GetCurrentProcess().Id`; this can be useful for printing the PID upon startup.
+
+Finally, launch `dotnet-counters` as follows:
+
+```console
+dotnet counters monitor Microsoft.EntityFrameworkCore -p <PID>
+```
+
+`dotnet-counters` will now attach to your running process and start reporting continuous counter data:
+
+```console
+Press p to pause, r to resume, q to quit.
+ Status: Running
+
+[Microsoft.EntityFrameworkCore]
+    Active DbContexts                                               1
+    Execution Strategy Operation Failures (Count / 1 sec)           0
+    Execution Strategy Operation Failures (Total)                   0
+    Optimistic Concurrency Failures (Count / 1 sec)                 0
+    Optimistic Concurrency Failures (Total)                         0
+    Queries (Count / 1 sec)                                         1
+    Queries (Total)                                               189
+    Query Cache Hit Rate (%)                                      100
+    SaveChanges (Count / 1 sec)                                     0
+    SaveChanges (Total)                                             0
+```
+
+## Counters and their meaning
+
+Counter name                          | Description
+------------------------------------- | ----
+Active DbContexts                     | The number of active, undisposed DbContext instances current in your application. If this number grows continuously, you may have a leak because DbContext instances aren't being properly disposed. Note that if [context pooling](xref:core/miscellaneous/context-pooling) is enabled, this number includes pooled DbContext instances not currently in use.
+Execution Strategy Operation Failures | The number of times a database operation failed to execute. If a retrying execution strategy is enabled, this includes each individual failure within multiple attempts on the same operation. This can be used to detect transient issues with your infrastructure.
+Optimistic Concurrency Failures       | The number of times 'SaveChanges' failed because of an optimistic concurrency error, because data in the data store was changed since your code loaded it. This corresponds to a <xref:Microsoft.EntityFrameworkCore.DbUpdateConcurrencyException> being thrown.
```

Comment:
```suggestion
Optimistic Concurrency Failures       | The number of times `SaveChanges` failed because of an optimistic concurrency error, because data in the data store was changed since your code loaded it. This corresponds to a <xref:Microsoft.EntityFrameworkCore.DbUpdateConcurrencyException> being thrown.
```

Reply:
Also in other places